### PR TITLE
add dump-key to tools/dht_put.cpp.

### PR DESCRIPTION
**Pull request by thomas-yuan**

### CODE REVIEW

File: tools/dht_put.cpp [hunk 1]
```diff
@@ -73,7 +74,7 @@ void usage()
 	exit(1);
 }
 
-alert* wait_for_alert(lt::session& s, int alert_type)
```
Comment (arvidn) [inline tools/dht_put.cpp]: there's actually a name clash here. in older versions of osx (and presumably bsd) there's a C function called session() (or maybe it's a struct) that gets pulled in via some system header. That's where this came from

Comment (thomas-yuan) [inline tools/dht_put.cpp]: ok, changed it back now.

File: tools/dht_put.cpp [hunk 2]
```diff
@@ -122,10 +123,105 @@ void put_string(entry& e, boost::array<char, 64>& sig, boost::uint64_t& seq
...
+	FILE* f = fopen(filename, "rb+");
+	if (f == NULL)
+	{
+		fprintf(stderr, "failed to open file \"%s\": (%d) %s\n"
+			, filename, errno, strerror(errno));
+		return 1;
+	}
+
+	unsigned char seed[32];
+	fread(seed, 1, 32, f);
```
Comment (arvidn) [inline tools/dht_put.cpp]: would you mind checking the return value here? (make sure it's 32, and fail if it isn't)

File: tools/dht_put.cpp [hunk 3]
```diff
@@ -122,10 +123,105 @@ void put_string(entry& e, boost::array<char, 64>& sig, boost::uint64_t& seq
...
+
+	FILE* f = fopen(filename, "wb+");
+	if (f == NULL)
+	{
+		fprintf(stderr, "failed to open file for writing \"%s\": (%d) %s\n"
+			, filename, errno, strerror(errno));
+		return 1;
+	}
+
+	fwrite(seed, 1, 32, f);
```
Comment (arvidn) [inline tools/dht_put.cpp]: same here, please make sure the return value is 32

File: tools/dht_put.cpp [hunk 4]
```diff
@@ -136,60 +232,34 @@ int main(int argc, char* argv[])
...
+		if (argc < 1) usage();
+
+		return generate_key(argv[0]);
 	}
 
 	settings_pack sett;
 	sett.set_bool(settings_pack::enable_dht, false);
 	sett.set_int(settings_pack::alert_mask, 0xffffffff);
-	lt::session s(sett);
+	session s(sett);
```
Comment (arvidn) [inline tools/dht_put.cpp]: I think the global session function affects this too

File: tools/dht_put.cpp [hunk 5]
```diff
@@ -136,60 +232,34 @@ int main(int argc, char* argv[])
...
 	s.add_dht_router(std::pair<std::string, int>("router.utorrent.com", 6881));
 	sett.set_bool(settings_pack::enable_dht, true);
 	s.apply_settings(sett);
 
-	FILE* f = fopen(".dht", "rb");
-	if (f != NULL)
-	{
-		fseek(f, 0, SEEK_END);
-		int size = ftell(f);
-		fseek(f, 0, SEEK_SET);
```
Comment (arvidn) [inline tools/dht_put.cpp]: I guess I'm guilty of the failure to check return value :(

File: tools/dht_put.cpp [hunk 6]
```diff
@@ -122,10 +123,105 @@ void put_string(entry& e, boost::array<char, 64>& sig, boost::uint64_t& seq
...
+	if (f == NULL) return;
+
+	fseek(f, 0, SEEK_END);
+	int size = ftell(f);
+	fseek(f, 0, SEEK_SET);
+
+	if (size > 0)
+	{
+		std::vector<char> state;
+		state.resize(size);
```
Comment (arvidn) [inline tools/dht_put.cpp]: session here too. I thought the travis osx build exposed this issue..

**Comment by thomas-yuan** [comment]
seems like the failed build is not caused by this patch.

**Pull request merged**